osu.Game/Beatmaps/BeatmapMetadata.cs

```diff
@@ -57,5 +58,18 @@ private BeatmapMetadata()
         IUser IBeatmapMetadataInfo.Author => Author;
 
         public override string ToString() => this.GetDisplayTitle();
+
+        public BeatmapMetadata DeepClone() => new BeatmapMetadata(Author.DeepClone())
```

Comment:
I'm not confident on us implementing these manually all over the place. One of the things I wanted to look at is replacing `IDeepColeable` with automapper. Is that something you could get behind?

That said, I think this is fine for the time being to get things working.

Reply:
Given the fact that I've messed up writing these clones no less than twice in this pull alone, you won't hear me protest against using automapper for these for sure.

Reply:
Just to confirm, the clones are to avoid reference issues at the detached side correct? And have nothing to do with realm.

Reply:
The metadata clone was originally to ensure 1:1 `BeatmapInfo` to `BeatmapMetadata` correspondence as you requested in an earlier version of this. It will not be actually necessary anymore for that original purpose, but probably safest to have for sanity to avoid reference issues as you say.

Reply:
If it's not required I'd argue we omit it to keep things as simple as possible... not sure.

Reply:
I'll also note that the metadata directly associated with the `BeatmapInfo` here will become managed upon being added, which is another reason to use a temporary clone for that.

I'll try removing but I feel somewhat safer with it.